giles-eco/src/main/java/edu/asu/diging/gilesecosystem/web/service/impl/IdentityProviderRegistry.java

```diff
@@ -44,8 +58,13 @@ public void addProvider(String providerId) {
     }
     
     @Override
-    public void addProviderTokenChecker(String providerId, String checkerId) {
-        providerTokenChecker.put(providerId, checkerId);
+    public void addProviderTokenChecker(String providerId, String authorizationType, String checkerId) {
+        if (authorizationType != null && !authorizationType.trim().isEmpty()) {
```

Comment:
why don't you do the if check in getProviderIdWithAuthType (just call it createProviderId) and then this is one line of code.